From 7967d5222d4f1478845a2cf2d661dd670a3e7b82 Mon Sep 17 00:00:00 2001
From: Xarblu <xarblu@protonmail.com>
Date: Wed, 5 Nov 2025 10:36:12 +0100
Subject: [PATCH] fix build with zlib-ng

minizip-ng dynamically includes
zlib-ng.h or zlib.h depending on
what was included prior

snes9x.h includes zlib.h
which needs to happen first or the
minizip include tries to include zlib-ng.h by default
resulting in an error because you can't
include both in a single translation unit
---
 conffile.h  | 2 +-
 loadzip.cpp | 2 +-
 memmap.cpp  | 2 +-
 stream.cpp  | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/conffile.h b/conffile.h
index d92128cba..ec3dc7cd4 100644
--- a/conffile.h
+++ b/conffile.h
@@ -12,6 +12,7 @@
 #include <vector>
 #include <string>
 
+#include "snes9x.h"
 #ifdef UNZIP_SUPPORT
 #  ifdef SYSTEM_ZIP
 #    include <minizip/unzip.h>
@@ -19,7 +20,6 @@
 #    include "unzip/unzip.h"
 #  endif
 #endif
-#include "snes9x.h"
 
 #ifndef MAX
 #  define MAX(a,b)  ((a) > (b)? (a) : (b))
diff --git a/loadzip.cpp b/loadzip.cpp
index 7082bb0e6..552b9fab5 100644
--- a/loadzip.cpp
+++ b/loadzip.cpp
@@ -8,12 +8,12 @@
 
 #include <assert.h>
 #include <ctype.h>
+#include "snes9x.h"
 #ifdef SYSTEM_ZIP
 #include <minizip/unzip.h>
 #else
 #include "unzip/unzip.h"
 #endif
-#include "snes9x.h"
 #include "memmap.h"
 
 
diff --git a/memmap.cpp b/memmap.cpp
index 1c483c6f4..3fba43daf 100644
--- a/memmap.cpp
+++ b/memmap.cpp
@@ -10,6 +10,7 @@
 #include <numeric>
 #include <assert.h>
 
+#include "snes9x.h"
 #ifdef UNZIP_SUPPORT
 #  ifdef SYSTEM_ZIP
 #    include <minizip/unzip.h>
@@ -25,7 +26,6 @@
 #include <ctype.h>
 #include <sys/stat.h>
 
-#include "snes9x.h"
 #include "memmap.h"
 #include "apu/apu.h"
 #include "fxemu.h"
diff --git a/stream.cpp b/stream.cpp
index d6e1c1cbb..173f5fd9a 100644
--- a/stream.cpp
+++ b/stream.cpp
@@ -7,6 +7,7 @@
 // Abstract the details of reading from zip files versus FILE *'s.
 
 #include <string>
+#include "snes9x.h"
 #ifdef UNZIP_SUPPORT
 #  ifdef SYSTEM_ZIP
 #    include <minizip/unzip.h>
@@ -14,7 +15,6 @@
 #    include "unzip.h"
 #  endif
 #endif
-#include "snes9x.h"
 #include "stream.h"
 
 
